.github/workflows/ui-tests-critical.yml

```diff
@@ -0,0 +1,101 @@
+name: UI Tests Critical
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    paths:
+      - 'Sources/**'
+      - '.github/workflows/ui-tests-critical.yml'
+      - 'fastlane/**'
+      - 'TestSamples/**'
+
+# https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  build-sample:
+    name: Build SwiftUITestSample Sample
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v4
+      - run: ./scripts/ci-select-xcode.sh 16
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - run: bundle exec fastlane build_ios_swift_ui_test_sample
+      - uses: actions/upload-artifact@v4
+        with:
+          name: SwiftUITestSample
+          path: TestSamples/SwiftUITestSample/DerivedData/Build/Products/Debug-iphonesimulator/SwiftUITestSample.app
+
+  run-tests:
+    name: Test iOS ${{matrix.os-version}} on ${{matrix.device}} Simulator
+    needs: [build-sample]
+    runs-on: ${{matrix.runs-on}}
+    env:
+      APP_ARTIFACT_NAME: "SwiftUITestSample"
+      APP_PATH: "Sample.app"
+
+      MAESTRO_FLOWS_PATH: "TestSamples/SwiftUITestSample/Maestro"
+      MAESTRO_LOGS_PATH: "MaestroLogs"
+
+      # https://github.com/facebook/react-native/blob/24e7f7d25629a7af6d877a0b79fed2faaab96437/.github/actions/maestro-ios/action.yml#L57
+      MAESTRO_DRIVER_STARTUP_TIMEOUT: 1500000 # 25 min, CI can be slow at times
+      MAX_ATTEMPTS: 3 # Maestro test suits re runs in case of flakyness
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - runs-on: macos-12
+            xcode: "13.4.1"
+            device: "iPhone 8"
+            os-version: "15.5"
+            create-simulator: true
+          - runs-on: macos-13
+            xcode: "14.3.1"
+            device: "iPhone 14"
+            os-version: "16.4"
+            create-simulator: true
+          - runs-on: macos-14-xlarge
```

Comment:
`m`: Please explain why you're going with xlarge here.

Reply:
Added comment, the same as the one in the PR here.